shaders-hlsl/geom/triangles.geom

```diff
@@ -0,0 +1,28 @@
+#version 450
+
+layout(points) in;
+layout(triangle_strip, max_vertices = 3) out;
+
+layout(location = 0) in vec4 vPositionIn[1];
+layout(location = 1) in vec3 vColorIn[1];
+
+layout(location = 0) out vec3 gColor;
+
+void main() {
```

Comment:
I'm missing a test for calling EmitVertex and  EndPrimitive from within a leaf function.

Reply:
Also, is inout for GeometryStream well defined in HLSL? I don't know of any alternative fwiw.

Reply:
> I'm missing a test for calling EmitVertex and EndPrimitive from within a leaf function.

Good catch. I didn't actually consider that - all the geometry shaders in my codebase are brutally inline.

> Also, is inout for GeometryStream well defined in HLSL? I don't know of any alternative fwiw.

https://microsoft.github.io/hlsl-specs/specs/hlsl.pdf is quiet on the topic.